src/WebJobs.Script/Config/WorkerConcurrencyOptionsSetup.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.Config
+{
+    internal class WorkerConcurrencyOptionsSetup : IConfigureOptions<WorkerConcurrencyOptions>
+    {
+        private readonly IConfiguration _configuration;
+        private readonly IEnvironment _environment;
+
+        public WorkerConcurrencyOptionsSetup(IConfiguration configuration, IEnvironment environment)
+        {
+            _configuration = configuration;
+            _environment = environment;
+        }
+
+        public void Configure(WorkerConcurrencyOptions options)
+        {
+            if (bool.TryParse(_environment.GetEnvironmentVariable(RpcWorkerConstants.FunctionWorkerDynamicConcurrencyEnabledSettingName), out bool enabled))
+            {
+                // Do not enable convurrency if any concurrency settings are defined
```

Comment:
Why can't we allow worker dynamic concurrency (WDC) enablement to override any other concurrency related settings? That way, customers who may have already configured things can try WDC by enabling a flag, without having to remove all their existing settings. This enables flipping back and forth via a single flag.

Reply:
We can have either WDC or custom concurrency settings, we do not want to allow customers to set WDC  + custom settings as it can make a confusion.